pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -7,65 +7,97 @@ import (
 	"k8s.io/apimachinery/pkg/api/resource"
 )
 
+type (
+	Flag            string
+	Count           string
+	PersistentCount string
+	Quantity        string
+
+	Flags            map[Flag]bool
+	Counts           map[Count]int
+	PersistentCounts map[PersistentCount]int
+	Quantities       map[Quantity]resource.Quantity
+)
+
+var (
+	Unlimited Flag = "unlimited"
+
+	Apps       Count = "apps"
+	Containers Count = "containers"
+	Jobs       Count = "jobs"
+	Images     Count = "images"
+	Projects   Count = "projects"
+
+	Volumes PersistentCount = "volumes"
+	Secrets PersistentCount = "secrets"
+
+	VolumeStorage Quantity = "volumeStorage"
+	Memory        Quantity = "memory"
+	CPU           Quantity = "cpu"
+)
+
 // Resources is a struct separate from the QuotaRequestInstanceSpec to allow for
 // external controllers to programmatically set the resources easier. Calls to
 // its functions are mutating.
 type Resources struct {
-	Unlimited bool `json:"unlimited,omitempty"`
-
-	Apps       int `json:"apps,omitempty"`
-	Containers int `json:"containers,omitempty"`
-	Jobs       int `json:"jobs,omitempty"`
-	Volumes    int `json:"volumes,omitempty"`
-	Secrets    int `json:"secrets,omitempty"`
-	Images     int `json:"images,omitempty"`
-	Projects   int `json:"projects,omitempty"`
-
-	VolumeStorage resource.Quantity `json:"volumeStorage,omitempty"`
-	Memory        resource.Quantity `json:"memory,omitempty"`
-	CPU           resource.Quantity `json:"cpu,omitempty"`
+	Flags            Flags            `json:"flags,omitempty"`
+	Counts           Counts           `json:"counts,omitempty"`
+	PersistentCounts PersistentCounts `json:"persistentCounts,omitempty"`
+	Quantities       Quantities       `json:"quantities,omitempty"`
+}
+
+// NewResources initializes and returns a new Resources instance with all maps initialized.
+func NewResources() Resources {
+	return Resources{
+		Flags:            make(Flags),
+		Counts:           make(Counts),
+		PersistentCounts: make(PersistentCounts),
+		Quantities:       make(Quantities),
+	}
 }
 
 // Add will add the resources of another Resources struct into the current one.
 func (current *Resources) Add(incoming Resources) {
-	current.Apps += incoming.Apps
-	current.Containers += incoming.Containers
-	current.Jobs += incoming.Jobs
-	current.Volumes += incoming.Volumes
-	current.Secrets += incoming.Secrets
-	current.Images += incoming.Images
-	current.Projects += incoming.Projects
-
-	current.VolumeStorage.Add(incoming.VolumeStorage)
-	current.Memory.Add(incoming.Memory)
-	current.CPU.Add(incoming.CPU)
+	// Add the incoming resources to the current ones
+	for flag, value := range incoming.Flags {
+		current.Flags[flag] = value
+	}
```

Comment:
This makes it possible for you to "add" incoming resources and flip unlimited to `false`. Is that the desired behavior?

Reply:
Yeah, I did this intentionally here. My thought was that there is some nice utility when you want to combine a bunch of resources together to be aggregated.

Reply:
So if one of a user's quota is unlimited and another one is not, then the aggregated quota could not be unlimited? The previous implementation was that if one of the quotas was unlimited, then the aggregated one would be. You're saying this was an intentional change?

Reply:
Oh I misread your comment, I'm sorry. I see what you're saying now and will fix it.

Reply:
Okay, I went ahead and removed it. I think it makes more sense as it was before than trying to clean-up the toggle-off scenario.